ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -443,12 +443,12 @@ public static BlobProperties buildBlobProperties(Map<String, Object> args) throw
           // key size
           sizeToAllocate += 4;
           String keyToStore = key.substring(Headers.USER_META_DATA_HEADER_PREFIX.length());
-          sizeToAllocate += keyToStore.getBytes(StandardCharsets.US_ASCII).length;
+          sizeToAllocate += keyToStore.getBytes(StandardCharsets.UTF_8).length;
           String value = getHeader(args, key, true);
           userMetadataMap.put(keyToStore, value);
           // value size
           sizeToAllocate += 4;
-          sizeToAllocate += value.getBytes(StandardCharsets.US_ASCII).length;
+          sizeToAllocate += value.getBytes(StandardCharsets.UTF_8).length;
```

Comment:
Can you make the Charset a class level constant?

Reply:
Sure, will do.